site/docs/plugins/i18n.md

```diff
@@ -1,105 +1,751 @@
+  return ctx.session.locale ??
+    ctx.from?.language_code ??
+    "en";
+}
+
+const i18n = new I18n({
+  localeNegotiator: myLocaleNegotiator,
+});
+```
+
+</CodeGroupItem>
+ <CodeGroupItem title="JavaScript">
+
+```ts
+function myLocaleNegotiator(ctx) {
+  return ctx.session.locale ??
+    ctx.from?.language_code ??
+    "en";
+}
+
+const i18n = new I18n({
+  localeNegotiator: myLocaleNegotiator,
+});
+```
+
+</CodeGroupItem>
+</CodeGroup>
+
+If the custom locale negotiator returns an invalid locale, it will fall back and choose a locale, following the above order.
+
+## Rendering Translated Messages
+
+Let's take a closer look at rendering messages.
+
+```ts
+bot.command("start", async (ctx) => {
+  // Call the "translate" or "t" helper to render the
+  // message by specifying its ID and additional parameters:
+  await ctx.reply(ctx.t("welcome"));
+});
+
+bot.command("cart", async (ctx) => {
+  // You can pass placeables as the second object.
+  await ctx.reply(ctx.t("cart-msg", { items: 10 }));
+});
+```
+
+Now you can `/start` your bot; it should render the following message:
+
+```:no-line-numbers
+Hi there!
+```
+
+Now, with the `/cart` command:
+
+```:no-line-numbers
+You currently have 10 items in your cart.
+```
+
+Try to change the value of the `items` variable to see how the rendered message would change!
+Also, check out the Fluent documentation, especially the [placeables documentation](https://projectfluent.org/fluent/guide/placeables.html).
+
+## Adding Translations
+
+There are three main methods to load translations.
+
+### Loading Locales Using the `directory` Option
+
+The simplest way to add translations to the `I18n` instance is by having all of your translations in a directory and specifying the directory name in the options.
+
+```ts
+const i18n = new I18n({
+  directory: "locales",
+});
+```
+
+### Loading Locales From a Directory
+
+This method is the same thing as specifying `directory` in options.
+Just put them all in a folder and load them like this:
+
+```ts
+const i18n = new I18n();
+
+await i18n.loadLocalesDir("locales");
+i18n.loadLocalesDirSync("locales 2");
+```
+
+### Loading a Single Locale
+
+It is also possible to add a single translation to the instance.
+You can either specify the file path to the translation using
+
+```ts
+const i18n = new I18n();
+
+await i18n.loadLocale("en", { filePath: "locales/en.ftl" });
+i18n.loadLocaleSync("de", { filePath: "locales/de.ftl" });
+```
+
+or you can directly load the translation data as a string like this:
+
+```ts
+const i18n = new I18n();
+
+await i18n.loadLocale("en", {
+  source: `greeting = Hello { $first_name }!
+language-set = Language has been set to English!`,
+});
+
+i18n.loadLocaleSync("de", {
+  source: `greeting = Hallo { $first_name }!
+language-set = Die Sprache wurde zu Deutsch geändert!`,
+});
+```
+
+## Custom Placeables and Functions
+
+These two features are going to help you when translating things.
+Fluent provides both.
```

Comment:
This sentence somehow does not make much sense to me here. We talk about two features of the plugin, but why do we say that Fluent provides both? Why else would we mention them, if they were not supported, eh?

Reply:
You're right, it doesn't make any sense. I don't know how it ended up like this. Originally this sentence wasn't there. I'll remove it.

Reply:
What do you think about replacing that paragraph with this:

This plugin can also help you with writing translations.

Reply:
Works!